status-im/status-react
# Remove support for editing shared addresses while request to join is pending approval

### Summary

This PR completely removes support for editing shared addresses while a user's request to join is pending approval.

The reason for the removal is that, in status-go, the control node that will process the message `protobuf.ApplicationMetadataMessage_COMMUNITY_EDIT_SHARED_ADDRESSES` verifies the user is already a member of the community, but a user waiting for approval to join is not yet a member. Removing this check doesn't sound like a good idea and we want to avoid bugs/complications for the next release. The feature isn't working anyway, so the purpose of this PR is also to cleanup code that doesn't work and won't be re-done anytime soon.

#### Areas that may be impacted

In theory none because we (devs & QAs) know enabling the feature flag doesn't cause regressions in other parts of the app.

### Steps to test

Dear QAs, in this PR, besides removing support for editing addresses while requests to join are pending, I'm also enabling by default the feature flag to edit addresses since the feature seems to be stable enough that the flag is unjustified now. This should facilitate testing on your part. If you have already tested editing shared addresses (I can't be sure now), then I simple smoke test should be enough because I didn't change the behavior for anything except requests to join in the pending state, which I'm sure you couldn't test because it never worked.

~Before merging this PR, and after you approve from a QA standpoint, I will remove the flag entirely from the code (it's quick & safe).~ Edit: We will only remove the flag once other unrelated problems are solved, which will allow QAs to thoroughly test the feature.

status: ready
